backend/src/apiserver/filter/filter_test.go

```diff
@@ -235,3 +236,46 @@ func TestAddToSelect(t *testing.T) {
 		}
 	}
 }
+
+func TestMarshalJSON(t *testing.T) {
+	f := &Filter{
+		filterProto: &api.Filter{
+			Predicates: []*api.Predicate{
+				&api.Predicate{
+					Key: "Name", Op: api.Predicate_EQUALS,
+					Value: &api.Predicate_StringValue{StringValue: "SomeName"},
+				},
+			},
+		},
+		eq: map[string]interface{}{"name": "SomeName"},
+	}
+
+	want := `{"FilterProto":"{\"predicates\":[{\"op\":\"EQUALS\",\"key\":\"Name\",\"stringValue\":\"SomeName\"}]}","EQ":{"name":"SomeName"},"NEQ":null,"GT":null,"GTE":null,"LT":null,"LTE":null,"IN":null,"SUBSTRING":null}`
```

Comment:
is it possible we omit the null?

Reply:
It is, if we add annotation to the struct. Since this is an internal struct, and explicit is better then implicit for debugging purposes, I left it as is. What do you think?